In [240]:
import numpy as np

n_x = 5
n_h = 3
n_y = 1

In [241]:
X = np.random.randn(n_x,1)
W1 = np.random.randn(n_h,n_x) 
b1 = np.zeros(shape=(n_h, 1))
W2 = np.random.randn(n_y,n_h) * 0.01
b2 = np.zeros(shape=(n_y, 1))

In [242]:
# Implement Forward Propagation to calculate A2 (probabilities)
import math

def sigmoid(x):
  return 1 / (1 + math.exp(-x))

def reluDerivative(x):
    x[x<=0] = 0
    x[x>0] = 1
    return x

Z1 = np.dot(W1,X) + b1
A1 = reluDerivative(Z1)
Z2 = np.dot(W2,A1) + b2
A2 = sigmoid(Z2) # Final 

In [243]:
print(A2)

0.5010301803582545


In [244]:
X_plug = [[1],[2],[3],[4],[5]]

W1_plug = [[.01, .01, .01, .01, .01], [.01, .01, .01, .01, .01], [.01, .01, .01, .01, .01]]

b1_plug = [[1],[1],[1]]

W2_plug = [[.02,.02,.02]]

b2_plug = [[1]]

In [245]:
# Implement Forward Propagation to calculate A2 (probabilities)
import math

def sigmoid(x):
  return 1 / (1 + math.exp(-x))

def reluDerivative(x):
    x[x<=0] = 0
    x[x>0] = 1
    return x

Z1_plug = np.dot(W1_plug,X_plug) + b1_plug
A1_plug = reluDerivative(Z1_plug)
Z2_plug = np.dot(W2_plug,A1_plug) + b2_plug
A2_plug = sigmoid(Z2_plug) # Final 

In [246]:
print(A2_plug)

0.7426905453026857


In [247]:
X_trial = [[1],[2],[3],[4],[5]]

W1_trial = [[.01, .01, .01, .01, .01], [.01, .01, .01, .01, .01], [.01, .01, .01, .01, .01]]

b1_trial = [[1],[1],[1]]

W2_trial = [[.02,.02,.02]]

b2_trial = [[-6.8]]

In [248]:
# Implement Forward Propagation to calculate A2 (probabilities)
import math

def sigmoid(x):
  return 1 / (1 + math.exp(-x))

def reluDerivative(x):
    x[x<=0] = 0
    x[x>0] = 1
    return x

Z1_trial = np.dot(W1_trial,X_trial) + b1_trial
A1_trial = reluDerivative(Z1_trial)
Z2_trial = np.dot(W2_trial,A1_trial) + b2_trial
A2_trial = sigmoid(Z2_trial) # Final 
print(A2_trial)

0.0011812501544791263


In [249]:
X_trial = [[1],[2],[3],[4],[5]]

W1_trial = [[.01, .01, .01, .01, .01], [.01, .01, .01, .01, .01], [.01, .01, .01, .01, .01]]

b1_trial = [[1],[1],[1]]

W2_trial = [[.02,.02,.02]]

b2_trial = [[-0.3]]

In [250]:
# Implement Forward Propagation to calculate A2 (probabilities)
import math

def sigmoid(x):
  return 1 / (1 + math.exp(-x))

def reluDerivative(x):
    x[x<=0] = 0
    x[x>0] = 1
    return x

Z1_trial = np.dot(W1_trial,X_trial) + b1_trial
A1_trial = reluDerivative(Z1_trial)
Z2_trial = np.dot(W2_trial,A1_trial) + b2_trial
A2_trial = sigmoid(Z2_trial) # Final 
print(A2_trial)

0.44028635073280703


## Note
By manually plugging the values of bias b2 from -6.8 to -0.3, we could change the output value from .001 to .44